File: examples/sector-scan.py
"""Scan an entire sector for opportunities."""
from veroq import VeroqClient

client = VeroqClient()
result = client.ask("Find undervalued tech stocks with strong momentum")

screener = result.get("data", {}).get("screener", {})
matches = screener.get("results", [])

if matches:
    print(f"Found {len(matches)} matches:\n")
    print(f"  {'Ticker':<8} {'Price':>8} {'RSI':>6} {'Change':>8} {'Signal':>8}")
    print(f"  {'-'*42}")
    for m in matches[:10]:
        ticker = m.get("ticker", "?")
        price = m.get("price", "?")
        rsi = m.get("rsi", "?")
        change = m.get("change_pct", "?")
        signal = m.get("signal", "?")
        print(f"  {ticker:<8} ${price:>7} {rsi:>6} {change:>7}% {signal:>8}")
    reasons = screener.get("match_reasons", {})
    if reasons:
        print("\nWhy these matched:")
        for ticker, reason in list(reasons.items())[:3]:
            print(f"  {ticker}: {reason}")
else:
    print(result.get("summary", "No sector scan results."))

File: examples/sector-scan.ts
// Scan an entire sector for opportunities
import { VeroqClient } from '@veroq/sdk';

const client = new VeroqClient();
const sector = process.argv[2] || 'semiconductors';
const result = await client.ask(`Scan the ${sector} sector — oversold stocks with strong fundamentals`);

console.log(result.summary);

if (result.endpoints_called?.length) {
  console.log(`\nEndpoints called: ${result.endpoints_called.length}`);
  console.log(`Credits used: ${result.credits_used}`);
}
